rootfs/etc/nginx/lua/balancer/sticky.lua

```diff
@@ -43,6 +44,20 @@ function _M.set_cookie(self, value)
     cookie_path = ngx.var.location_path
   end
 
+  local cookie_samesite = self.cookie_session_affinity.samesite
+  if cookie_samesite then
+    local cookie_conditional_samesite_none = self.cookie_session_affinity.conditional_samesite_none
+    if cookie_conditional_samesite_none
+        and cookie_samesite == "None"
+        and not same_site.same_site_none_compatible(ngx.var.http_user_agent) then
```

Comment:
Looking at https://github.com/cloudflare/lua-resty-cookie/commit/8f414390cc72da90e9a450a2127acaea247e3dda#diff-8944ed02502a8a0a1f17e24e9b28c7a7R145, it seems like the library we use won't let us to pass `"None"` for `samesite`.  But given the new Chrome change you linked we should be able to set `samesite` to `"Nonde"`.

I think I'd rather fix this in the upstream library than hack it here.

Therefore in the meantime I suggest we comply with what the upstream library provides, i.e not support `"None"` value at all. Then we patch the upstream library to support this new Chrome change and come back here to reflect that.

Reply:
Yes, I'd rather fix it upstream as well. There is a currently a PR to that library to add support for SameSite=None, added 11 days ago. https://github.com/cloudflare/lua-resty-cookie/pull/32

Unfortunately the library hasn't been updated in 3 years. Of the 5 contributors, only 1 has an active current github history.

According to Google Chrome's pages, version 80 is set to ship Feb 4th with 50% of users being on the new SameSite settings. I'm not sure we can wait around for a PR to go through on a repo that hasn't been touched in 3 years. I'm happy to come back through and change this to use the samesite argument in the library once it accepts it. What do you think? 

https://www.chromestatus.com/features/schedule
https://www.chromium.org/updates/same-site